Update architecture doc page. (#1076)
quickwit-inc/quickwit: docs/design/architecture.md

sidebar_position: 1
---

> Quickwit has been designed from the ground up for cost-efficiency, scalability, and cloud-friendly. 

Reviewer: ```suggestion
Quickwit has been designed from the ground up for cost-efficiency and scalability. 
```

We can keep cloud-friendliness, but as is, this is an adjective within a list of nouns

> Quickwit core relies on two processes:

Reviewer: I wonder if we should say two `types of processes`. Unsure.

Author: Yes that’s what I wanted to say actually

- The Searchers for executing search queries from the REST API.

> - And the Indexers that index data from data sources.

Reviewer: ```suggestion
- The Indexers that index data from data sources.
```

Moreover, Quickwit leverages existing infrastructure by relying on battled-tested technologies for index storage, metadata storage, and ingestion:

### Index storage

> Quickwit stores indexed data on Amazon S3. In a single-server deployment, this can also be your local disk. We plan to support HDFS or any other cloud storage.

Reviewer: ```suggestion
Quickwit stores its indexes's file on Amazon S3. In a single-server deployment, this can also be your local disk. We plan to support HDFS or any other cloud storage.
```

> Quickwit stores indexed data on Amazon S3. In a single-server deployment, this can also be your local disk. We plan to support HDFS or any other cloud storage.

Reviewer: ```suggestion
Quickwit stores indexed data on Amazon S3. In a single-server deployment, this can also be your local disk. We plan to support other cloud storage, as well as HDFS.
```
